### ISSUE TITLE
[Docs]: cloudwatch_log_resource_policy - Could not convert to persistable policy

### ISSUE THREAD

**Issue by FransUrbo** [hashicorp/terraform-provider-aws#32260]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_web_acl_logging_configuration

### Description

Following the CloudWatch example, I get an error when applying it:

Error: creating CloudWatch Logs Resource Policy (torx-tf-waf-cf_logging): InvalidParameterException: Could not convert to persistable policy.

Not sure if this is a bug in the AWS provider or the documentation, but I report it as a documentation issue..

### Would you like to implement a fix?

None

**Comment by FransUrbo** [comment]
The problem seems to be in the `principals` section:
```
    principals {
      identifiers = ["delivery.logs.amazonaws.com"]
      type        = "AWS"
    }
```
Changing the type from `AWS` to `Service` worked.

**Comment by FransUrbo** [comment]
This is correct in the `cloudwatch_log_resource_policy` documentation: 

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_resource_policy

**Comment by camway** [comment]
Thank you! I was just writing up this bug report, and decided to do one last search for the error which lead me here. This seems to have resolved it for me.

**Comment by justinretzolk** [comment]
Hey @FransUrbo 👋 Thank you for taking the time to raise this! It looks like this was fixed by #32570 a couple of weeks ago, and is now live in the Registry 🎉. Given that, I'll go ahead and close this issue out. Thanks again for helping us to keep improving the docs!
